onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -2267,11 +2268,14 @@ Status DeviceBasedPartitioner::PartitionGraph(const onnxruntime::GraphViewer& gr
         device_types_.push_back(device_type);
         it = device_to_stream.find(device_type);
       }
-      // put the node into the belonging stream
-      if (node_name.empty()) {
-        node_names_by_stream_[it->second].push_back(op_type + std::to_string(op_type_counter[op_type]++));
+      std::string node_name_or_type = node_name;
+      if (node_name_or_type.empty()) {
+        node_name_or_type = op_type + std::to_string(op_type_counter[op_type]++);
+      }
+      if (op_type == "MemcpyToHost") {
+        index_of_MemcpyToHost.push_back(node->Index());
```

Comment:
besides the update in partition graph, could we also simplify the notification/wait generation because now we avoid the within-stream-cross-devie data transfer?

Reply:
We still face the chance of this case, for example, the user may use partition JSON file to explicitly make MemcpyToHost and its consumer in the same stream (Actually there is a test case PlannerTest.MultiStreamCudaEPNodeCPUOutput for this case). Let me leave a comment in the code for the future reference